tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteConsumerGroupsTest.java

```diff
@@ -17,279 +17,374 @@
-            "The consumer group deletion did not work as expected");
+    private String composeGroupId(GroupProtocol protocol) {
+        String groupPrefix = "test.";
+        return protocol != null ? groupPrefix + protocol.name : groupPrefix + "dummy";
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testDeleteWithMixOfSuccessAndError(String quorum) throws Exception {
-        createOffsetsTopic(listenerName(), new Properties());
-        String missingGroup = "missing.group";
+    private String composeTopicName(GroupProtocol protocol) {
+        String topicPrefix = "foo.";
+        return protocol != null ? topicPrefix + protocol.name : topicPrefix + "dummy";
+    }
 
-        // run one consumer in the group
-        ConsumerGroupExecutor executor = addConsumerGroupExecutor(1);
-        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP};
-        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+    private String composeMissingGroupId(GroupProtocol protocol) {
+        String missingGroupPrefix = "missing.";
+        return protocol != null ? missingGroupPrefix + protocol.name : missingGroupPrefix + "dummy";
+    }
 
-        TestUtils.waitForCondition(
-            () -> service.listConsumerGroups().contains(GROUP) && Objects.equals(service.collectGroupState(GROUP).state, "Stable"),
-            "The group did not initialize as expected.");
+    private AutoCloseable consumerGroupClosable(GroupProtocol protocol, String groupId, String topicName) {
+        Map<String, Object> configs = composeConfigs(
+                groupId,
+                protocol.name,
+                emptyMap()
+        );
 
-        executor.shutdown();
+        return ConsumerGroupExecutor.buildConsumers(
+                1,
+                false,
+                topicName,
+                () -> new KafkaConsumer<String, String>(configs)
+        );
+    }
 
-        TestUtils.waitForCondition(
-            () -> Objects.equals(service.collectGroupState(GROUP).state, "Empty"),
-            "The group did not become empty as expected.");
+    private Predicate<String> predicateGroupState(ConsumerGroupCommand.ConsumerGroupService service, ConsumerGroupState state) {
+        return groupId -> {
+            try {
+                return checkGroupState(service, groupId, state);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        };
+    }
 
-        cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP, "--group", missingGroup};
+    private boolean checkGroupState(ConsumerGroupCommand.ConsumerGroupService service, String groupId, ConsumerGroupState state) throws Exception {
+        return Objects.equals(service.collectGroupState(groupId).state, state.toString());
```

Comment:
It seems to me `state` should keep the origin type instead of `String`. Could you revise that and update the test here? 

https://github.com/apache/kafka/blob/9b8aac22ec7ce927a2ceb2bfe7afd57419ee946c/tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommand.java#L781